lms/djangoapps/grades/admin/forms.py

```diff
@@ -0,0 +1,40 @@
+"""
+Defines a form for providing validation of subsection grade templates.
+"""
+import logging
+
+from django import forms
+
+from lms.djangoapps.grades.admin.models import CoursePersistentGradesFlag
+
+from opaque_keys import InvalidKeyError
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.locator import CourseLocator
+
+log = logging.getLogger(__name__)
+
+
+class CoursePersistentGradesAdminForm(forms.ModelForm):
+    """Input form for subsection grade enabling, allowing us to verify data."""
+
+    class Meta(object):
+        model = CoursePersistentGradesFlag
+        fields = '__all__'
+
+    def clean_course_id(self):
+        """Validate the course id"""
+        cleaned_id = self.cleaned_data["course_id"]
+        try:
+            course_key = CourseLocator.from_string(cleaned_id)
+        except InvalidKeyError:
+            msg = u'Course id invalid. Entered course id was: "{0}."' \
+                  u' Please recheck that you have supplied a valid course id.'.format(cleaned_id)
+            raise forms.ValidationError(msg)
+
+        if not modulestore().has_course(course_key):
+            msg = u'COURSE NOT FOUND'
+            msg += u' --- Entered course id was: "{0}". '.format(course_key.to_deprecated_string())
+            msg += 'Please recheck that you have supplied a valid course id.'
```

Comment:
Looking at this, like above, I don't think this entire message is necessary. The user just needs to know that the course wasn't found.